### PR TITLE
feat: Imperative `setTarget` for `usePopper` and `positioning`

### PR DESCRIPTION
**Pull request by ling1726**
## Current Behavior

Using `<Menu positioning={{ target }} />` requires `React.useState()` (to get refs as DOM elements) call to force a double render.

## New Behavior

Adds a `setTarget` imperative call to the `popperRef` imperative handle.

This allows users to set their own target **without** forcing a double render with `React.useState()`. `setTarget()` can be called inside `React.useEffect()` hooks, for example. The modified stories all use effects to call this API.

`onStateUpdate()` was removed `usePopper()`, as we don't it publicly it's not a breaking change. Types (`PopperOptions`, `PositioningOptions`) were also refactored to better show boundaries  between the hook and options themselves. No breaking changes in these interfaces.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f30f8424f3d0b4e720c30583bddef58625263d33'>f30f8424f3d0b4e720c30583bddef58625263d33</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount_base.html">16809</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount.html">8891</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount_base.html">2157</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount.html">4279</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Avatar-mount_base.html">691</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Avatar-mount.html">688</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/BaseButton-mount_base.html">812</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/BaseButton-mount.html">800</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Breadcrumb-mount_base.html">2307</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Breadcrumb-mount.html">2332</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ButtonNext-mount_base.html">408</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ButtonNext-mount.html">435</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Checkbox-mount_base.html">1315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Checkbox-mount.html">1348</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CheckboxBase-mount_base.html">1119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CheckboxBase-mount.html">1148</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ChoiceGroup-mount_base.html">6886</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ChoiceGroup-mount.html">4084</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ComboBox-mount_base.html">823</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ComboBox-mount.html">838</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount_base.html">16809</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount.html">8891</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/CommandBar-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ContextualMenu-mount_base.html">7194</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ContextualMenu-mount.html">7273</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DefaultButton-mount_base.html">988</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DefaultButton-mount.html">1011</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRow-mount_base.html">3315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRow-mount.html">8205</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRowFast-mount_base.html">3212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRowFast-mount.html">3147</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRowNoStyles-mount_base.html">3033</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DetailsRowNoStyles-mount.html">4117</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Dialog-mount_base.html">2217</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Dialog-mount.html">2224</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DocumentCardTitle-mount_base.html">151</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/DocumentCardTitle-mount.html">165</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Dropdown-mount_base.html">2759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Dropdown-mount.html">2749</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderNext-mount_base.html">1679</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderNext-mount.html">1763</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-mount_base.html">126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-mount.html">154</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-virtual-rerender_base.html">99</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-virtual-rerender.html">280</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">143</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount.html">177</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FocusTrapZone-mount_base.html">1612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FocusTrapZone-mount.html">1603</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FocusZone-mount_base.html">2554</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/FocusZone-mount.html">1587</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/IconButton-mount_base.html">1522</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/IconButton-mount.html">1533</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Label-mount_base.html">317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Label-mount.html">335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Layer-mount_base.html">3354</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Layer-mount.html">2592</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Link-mount_base.html">425</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Link-mount.html">430</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MakeStyles-mount_base.html">1494</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MakeStyles-mount.html">2473</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MenuButton-mount_base.html">1276</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MenuButton-mount.html">1263</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MessageBar-mount_base.html">1740</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/MessageBar-mount.html">1753</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Nav-mount_base.html">2814</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Nav-mount.html">2835</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/OverflowSet-mount_base.html">983</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/OverflowSet-mount.html">2825</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount_base.html">2157</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount.html">4279</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Panel-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Persona-mount_base.html">2131</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Persona-mount.html">742</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Pivot-mount_base.html">2225</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Pivot-mount.html">1244</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/PrimaryButton-mount_base.html">1120</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/PrimaryButton-mount.html">1134</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Rating-mount_base.html">6571</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Rating-mount.html">6602</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SearchBox-mount_base.html">1187</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SearchBox-mount.html">1162</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Shimmer-mount_base.html">2239</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Shimmer-mount.html">2201</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Slider-mount_base.html">1670</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Slider-mount.html">1703</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SpinButton-mount_base.html">4285</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SpinButton-mount.html">6129</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Spinner-mount_base.html">1190</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Spinner-mount.html">398</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SplitButton-mount_base.html">2694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SplitButton-mount.html">6460</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Stack-mount_base.html">465</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Stack-mount.html">476</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/StackWithIntrinsicChildren-mount_base.html">2005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/StackWithIntrinsicChildren-mount.html">1942</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/StackWithTextChildren-mount_base.html">4575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/StackWithTextChildren-mount.html">4534</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SwatchColorPicker-mount_base.html">10320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/SwatchColorPicker-mount.html">9651</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TagPicker-mount_base.html">2295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TagPicker-mount.html">2335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TeachingBubble-mount_base.html">11288</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TeachingBubble-mount.html">11261</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Text-mount_base.html">393</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Text-mount.html">391</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TextField-mount_base.html">1234</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/TextField-mount.html">1197</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-mount_base.html">1047</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-mount.html">1061</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-virtual-rerender_base.html">556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-virtual-rerender.html">544</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1600</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1612</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Toggle-mount_base.html">716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/Toggle-mount.html">713</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/buttonNative-mount_base.html">133</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21632/perf-test/buttonNative-mount.html">135</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

## Perf Analysis (`@fluentui/react-northstar`)
:warning: No perf measurements available
<div id="perfComment9423"/>

File: packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx [hunk 1]
```diff
@@ -3,23 +3,40 @@ import * as React from 'react';
...
     setOpen(data.open);
   };
 
+  React.useEffect(() => {
+    if (buttonRef.current) {
+      popperRef.current?.setTarget(buttonRef.current);
+    }
+  }, [buttonRef, popperRef]);
+
+  // A callback ref can be used, but the component the ref is attached to
```
Comment (ling1726) [inline packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx:22]: not sure about this one, with the limitations on renderOrder for the imperative handle to be assigned, should we just not encourage this at all ?

File: packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx [hunk 2]
```diff
@@ -3,23 +3,40 @@ import * as React from 'react';
 import { Menu, MenuList, MenuItem, MenuPopover, MenuProps } from '../index';
 
 import { Button } from '@fluentui/react-button';
+import { PopperRefHandle } from '@fluentui/react-positioning';
 
 export const AnchorToCustomTarget = () => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const popperRef = React.useRef<PopperRefHandle>(null);
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx:10]: created an issue for this to follow up #21669

File: packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx [hunk 3]
```diff
@@ -3,23 +3,40 @@ import * as React from 'react';
...
+  // A callback ref can be used, but the component the ref is attached to
+  // must be rendered after the `Menu` component so that `popperRef` is resolved
+  // Most likely won't work if there's any kind of async loading
+  // const callbackRef = React.useCallback(
+  //   (el: HTMLButtonElement) => {
+  //     console.log(popperRef.current, el);
+  //     popperRef.current?.setTarget(el);
+  //   },
+  //   [popperRef],
+  // );
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx:31]: yeah it was also a concern from a self comment, I've removed this

**Pull request merged**